pkg/endpoint/local_ip.go

```diff
@@ -29,27 +32,40 @@ var familyToDestIP = map[k8snet.IPFamily]string{
 	k8snet.IPv6: "2001:4860:4860::8888",
 }
 
-func getLocalIPv4ForDestination(dst string) string {
-	conn, err := net.Dial("udp4", dst+":53")
-	logger.FatalOnError(err, "Error getting local IP")
+func getLocalIPFromRoutes(family string) (string, error) {
+	familyInt := netlink.FAMILY_V4
+	if family == string(k8snet.IPv6) {
+		familyInt = netlink.FAMILY_V6
+	}
 
-	defer conn.Close()
+	routes, err := netlink.RouteList(nil, familyInt)
```

Comment:
Use the `submariner-io/submariner/pkg/netlink` wrapper so this can be mocked in unit tests.